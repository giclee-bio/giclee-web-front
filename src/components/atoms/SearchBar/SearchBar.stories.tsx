import { Meta, StoryObj } from '@storybook/react'

import SearchBar from './SearchBar'

const meta: Meta<typeof SearchBar> = {}

export default meta
type Story = StoryObj<typeof SearchBar>

export const Index: Story = {
  args: {},
}
