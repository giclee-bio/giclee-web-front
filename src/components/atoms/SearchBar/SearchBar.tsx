'use client'

import * as Form from '@radix-ui/react-form'
import { MagnifyingGlassIcon } from '@radix-ui/react-icons'

const SearchBar = () => {
  return (
    <Form.Root>
      <Form.Field name='search'>
        <Form.Label className='sr-only'>Search</Form.Label>
        <div className='relative flex h-8'>
          <MagnifyingGlassIcon
            className='absolute left-1 h-8 text-stone-400'
            height='24'
            width='24'
          />
          <Form.Control asChild>
            <input
              className='rounded bg-stone-50 pl-8 pr-3 text-black placeholder:text-sm placeholder:text-stone-200'
              placeholder='さがす'
              required
              type='search'
            />
          </Form.Control>
        </div>
      </Form.Field>
    </Form.Root>
  )
}

export default SearchBar
