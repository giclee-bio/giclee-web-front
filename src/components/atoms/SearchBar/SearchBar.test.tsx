import { render } from '@testing-library/react'

import SearchBar from './SearchBar'

describe('SearchBar', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<SearchBar />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
