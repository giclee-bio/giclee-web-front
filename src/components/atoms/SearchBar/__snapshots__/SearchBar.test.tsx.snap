// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        class="sr-only"
        for="radix-:r0:"
      >
        Search
      </label>
      <div
        class="relative flex h-8"
      >
        <svg
          class="absolute left-1 h-8 text-stone-400"
          fill="none"
          height="24"
          viewBox="0 0 15 15"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 6.5C10 8.433 8.433 10 6.5 10C4.567 10 3 8.433 3 6.5C3 4.567 4.567 3 6.5 3C8.433 3 10 4.567 10 6.5ZM9.30884 10.0159C8.53901 10.6318 7.56251 11 6.5 11C4.01472 11 2 8.98528 2 6.5C2 4.01472 4.01472 2 6.5 2C8.98528 2 11 4.01472 11 6.5C11 7.56251 10.6318 8.53901 10.0159 9.30884L12.8536 12.1464C13.0488 12.3417 13.0488 12.6583 12.8536 12.8536C12.6583 13.0488 12.3417 13.0488 12.1464 12.8536L9.30884 10.0159Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <input
          class="rounded bg-stone-50 pl-8 pr-3 text-black placeholder:text-sm placeholder:text-stone-200"
          id="radix-:r0:"
          name="search"
          placeholder="さがす"
          required=""
          title=""
          type="search"
        />
      </div>
    </div>
  </form>
</DocumentFragment>
`;
